pkg/controllers/nodeclaim/termination/metrics.go

```diff
@@ -0,0 +1,39 @@
+/*
+Copyright The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package termination
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	crmetrics "sigs.k8s.io/controller-runtime/pkg/metrics"
+
+	"sigs.k8s.io/karpenter/pkg/metrics"
+)
+
+func init() {
+	crmetrics.Registry.MustRegister(NodeClaimTerminationDuration)
+}
+
+var NodeClaimTerminationDuration = prometheus.NewHistogramVec(
+	prometheus.HistogramOpts{
+		Namespace: metrics.Namespace,
+		Subsystem: metrics.NodeClaimSubsystem,
+		Name:      "termination_duration_seconds",
+		Help:      "Duration of nodeClaim termination in seconds.",
```

Comment:
```suggestion
		Help:      "Duration of NodeClaim termination in seconds.",
```

nit

Reply:
Is it termination of the NodeClaim or termination of the CloudProvider instance? I want to try to differentiate between the entire NodeClaim termination time and the specific time it takes to terminate the cloudprovider instance

Reply:
As of now, this metric is being fired when the finalizer from nodeClaim is removed and we make a kubeclient.Update call that succeeds. Do you think I should fire it when the underlying instance is gone to record the termination of the CloudProvider instance

Reply:
Yes, I think the thing that we are trying to represent here is the time between when we start terminating the CloudProvider _instance_ and when it actually terminates on the CloudProvider's side. Ideally, our metrics allow us to see:
1. The total termination time of the NodeClaim
2. The time within the termination time that it took to drain the Node
3. The time within the termination time that it took to terminate the instance

We should validate that we get each of these after your chagne